app/bundles/LeadBundle/Entity/TagRepository.php

```diff
@@ -91,18 +89,38 @@ public function checkLeadByTags(Lead $lead, $tags)
 
         $q = $this->_em->getConnection()->createQueryBuilder();
         $q->select('l.id')
-            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l');
-        $q->join('l', MAUTIC_TABLE_PREFIX.'lead_tags_xref', 'x', 'l.id = x.lead_id')
-                ->join('l', MAUTIC_TABLE_PREFIX.'lead_tags', 't', 'x.tag_id = t.id')
-                ->where(
-                    $q->expr()->andX(
-                        $q->expr()->in('t.tag', ':tags'),
-                        $q->expr()->eq('l.id', ':leadId')
-                    )
+            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l')
+            ->join('l', MAUTIC_TABLE_PREFIX.'lead_tags_xref', 'x', 'l.id = x.lead_id')
+            ->join('l', MAUTIC_TABLE_PREFIX.'lead_tags', 't', 'x.tag_id = t.id')
+            ->where(
+                $q->expr()->andX(
+                    $q->expr()->in('t.tag', ':tags'),
+                    $q->expr()->eq('l.id', ':leadId')
                 )
-                ->setParameter('tags', $tags, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY)
-                ->setParameter('leadId', $lead->getId());
+            )
+            ->setParameter('tags', $tags, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY)
+            ->setParameter('leadId', $lead->getId());
 
         return (bool) $q->execute()->fetchColumn();
     }
+
+    /**
+     * Save an entity through the repository.
+     *
+     * @param object $entity
+     * @param bool   $flush  true by default; use false if persisting in batches
+     *
+     * @return int
+     */
+    public function saveEntity($entity, $flush = true)
+    {
+        if (!$entity->getId() && $existingTag = $this->findOneBy(['tag' => $entity->getTag()])) {
+            // Do not save if the tag exists and instead add the ID of the existing tag to the entity
+            $entity->setId($existingTag->getId());
```

Comment:
What is the idea of this code? We cannot set ID of any entity if we use `autoincrement` functionality.
This code also means that you want to have 2 entites with same ID. That is really confusing. What is the reason?

Reply:
The reason is that I don't want to create duplicate tags. To accomplish that I search for the tag before save. If it exists, I don't need to save it, but I do need to update the entity ID because the rest of the code expects the entity will have an ID after calling the `saveEntity` method.

And if I know the tag label and ID, I don't have to continue saving it since it's all the entity has.